packages/x-components/src/x-modules/history-queries/store/actions/update-history-queries-with-search-response.action.ts

```diff
@@ -0,0 +1,46 @@
+import { HistoryQueriesXStoreModule } from '../types';
+
+/**
+ * Default implementation for the
+ * {@link HistoryQueriesActions.updateHistoryQueriesWithSearchResponse} action.
+ *
+ * The matching history query will only be updated on the following scenarios:
+ * 1. If it is part of a previous session, not the current one.
+ * 2. If its total results count has not been registered yet.
+ * 3. If its total results count registered is less than the one specified on the search response,
+ * meaning that the previous update was part of a filtered request.
+ *
+ * @param context - The {@link https://vuex.vuejs.org/guide/actions.html | context} of the actions,
+ * provided by Vuex.
+ *
+ * @param searchResponse - The search response to update history queries with.
+ *
+ * @returns A `void` promise that resolves when the history query finishes updating.
+ *
+ * @public
+ */
+// eslint-disable-next-line max-len
+export const updateHistoryQueriesWithSearchResponse: HistoryQueriesXStoreModule['actions']['updateHistoryQueriesWithSearchResponse'] =
+  ({ state, dispatch }, searchResponse) => {
+    if (searchResponse.status === 'success') {
+      const indexOfHistoryQuery = state.historyQueries.findIndex(
+        ({ query }) => query === searchResponse.request.query
+      );
+      if (indexOfHistoryQuery >= 0) {
+        const historyQuery = state.historyQueries[indexOfHistoryQuery];
+        const isCurrentSessionHistoryQuery = historyQuery.timestamp > state.sessionTimeStampInMs;
```

Comment:
Now that I think about this, shouldn't this be an equal or greater condition?

```suggestion
        const isCurrentSessionHistoryQuery = historyQuery.timestamp >= state.sessionTimeStampInMs;
```

The `sessionTimeStampInMs` contains the timestamp when the current session started, so If the query timestamp is equal it was done in the current session.

Reply:
Mmm I took the `sessionHistoryQueries` as a reference, there the condition is `>`, don't you think here it should be the same? @ajperezbau

Reply:
You are right!
It should be the same, so it is ok as it is 👌